Add Support for the NW Digital Radio DRAWS/UDRC

mcdermj opened the pull request:
This series adds support for the Northwest Digital Radio UDRC and DRAWS HATs.  The tlv320aic32x4 driver changes have already been committed upstream and are marked as such.  The dtoverlay pieces are particular to this project and are included here.  The board is capable of using the ASoC "simplecard" driver and the DTs reflect that.

mcdermj committed:
dtoverlays: Add Support for the UDRC/DRAWS

Adds a new overlay to support the Northwest Digital Radio
DRAWS and UDRC HATs.  See http://nwdigitalradio.com.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>

mcdermj committed:
ASoC: tlv320aic32x4: Fix Common Pins

commit c63adb28f6d913310430f14c69f0a2ea55eed0cc upstream.

The common pins were mistakenly not added to the DAPM graph.
Adding these pins will allow valid graphs to be created.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
Signed-off-by: Mark Brown <broonie@kernel.org>

mcdermj committed:
ASoC: tlv320aic32x4: Add Switch for Setting Common Mode Voltage

commit 44ceee847e27c828f2f1ef4e400e6bc0c8d04de3 upstream.

Add a switch for setting common mode voltage.  This can allow
for higher drive levels on the amplifier outputs.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
Signed-off-by: Mark Brown <broonie@kernel.org>

mcdermj committed:
ASoC: tlv320aic32x4: Add Playback PowerTune Controls

commit d3e6e374566e1154820a9a3dc82f7eef646fcf95 upstream.

PowerTune controls the power level of the chip.  On playback this
indirectly controls things like the gain of the various output
amplifiers.  This can allow for the decrease of output levels
from the codec.  This adds controls for those power levels to
the driver.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
Signed-off-by: Mark Brown <broonie@kernel.org>

mcdermj committed:
ASoC: tlv320aic32x4: Change author's name

commit 7297ba6c74c5b9e78d8e936af82eecfcf7d32dfb upstream.

The author of these files has changed her name.  Update
instances in the code of her dead name to current legal
name.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
Signed-off-by: Mark Brown <broonie@kernel.org>

mcdermj committed:
ASoC: tlv320aic32x4: Update copyright and use SPDX identifier

commit 8a1d95c393d971e624fc28f11516b0bc3a7fa706 upstream.

Update the copyright dates and use the SPDX identifier instead
of reciting the license.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
Signed-off-by: Mark Brown <broonie@kernel.org>

pelwell commented:
The `_gain` and `_datarate` parameters have been declared as targeting strings when they're actually targeting 32-bit integers; you need to but `:0` (4-byte integer at byte offset 0) after the property name in the parameter declarations:
```
        draws_adc_ch4_gain = <&adc_ch4>,"ti,gain:0";
```
I'd also like an entry in the README in the overlays directory, with a brief description of the overlay and its parameters - beware the strict formatting rules: no tabs, fixed indents and two blank lines between overlays.

Otherwise, nice job.

mcdermj committed:
dtoverlays: Change override targets to integers

Per Phil Elwell the gain and datarate parameters need to be
marked as integers rather than strings.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>

mcdermj committed:
dtoverlays:  Provide proper README entries for new boards

Per Phil Elwell, provide descriptions of the new overlays in the
README documentation.

Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>

mcdermj commented:
Thanks so much for your patience and commentary, Phil.  I have fixed your concerns and pushed  a couple of new commits that should solve the problems.

pelwell commented:
A single TAB crept into the README (editors can be sneaky), and the extra commits were really fixups to the overlay commit, so they've been squashed and merged offline.
